curriculum/challenges/english/25-front-end-development/lab-blog-post-card/66eaddd04a9e533fba689001.md

```diff
@@ -0,0 +1,393 @@
+```
+
+You should target `.blog-post-card` and set its `text-align` property.
+
+```js
+assert.isNotEmpty(new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("text-align"));
+```
+
+You should target `.blog-post-card` and set its shadow property.
+
+```js
+const card = document.querySelector('.blog-post-card');
+
+function hasBoxShadow(element) {
+  const boxShadow = getComputedStyle(element).getPropertyValue('box-shadow');
+  return boxShadow && boxShadow !== 'none';
+}
+
+assert.isTrue(hasBoxShadow(card));
+```
+
+You should target `.blog-post-card` and set its `overflow` property to `hidden`.
+
+```js
+const overflowVal = new __helpers.CSSHelp(document).getStyle(".blog-post-card")?.getPropertyValue("overflow");
+assert.equal(overflowVal, "hidden");
+```
+
+You should target `.post-content` and set its `padding` property.
+
+```js
+assert.isNotEmpty(new __helpers.CSSHelp(document).getStyle(".post-content")?.getPropertyValue("padding"));
+```
+
+`.read-more` should have a hover effect.
+
+```js
+const link = document.querySelector('.read-more');
+assert.exists(link);
+
+const stylesheet = [...document.styleSheets].find(sheet => [...sheet.rules].some(rule => rule.selectorText === '.read-more:hover'));
+assert.exists(stylesheet);
+
+const hoverRule = [...stylesheet.rules].find(rule => rule.selectorText === '.read-more:hover');
+const hoverBackgroundColor = hoverRule.style.backgroundColor;
+assert.isNotEmpty(hoverBackgroundColor);
+```
+
+`.read-more` should be styled like a button. 
+
+```js
+const readMore = document.querySelector('.read-more');
+assert.exists(readMore);
+
+const readMoreDisplay = getComputedStyle(readMore).display;
+assert.strictEqual(readMoreDisplay, 'inline-block');
+
+const readMoreMargin = getComputedStyle(readMore).marginTop;
+assert.notStrictEqual(readMoreMargin, '0px');
+
+const readMorePadding = getComputedStyle(readMore).padding;
+assert.notStrictEqual(readMorePadding, '0px');
+
+const readMoreBackgroundColor = getComputedStyle(readMore).backgroundColor;
+assert.notStrictEqual(readMoreBackgroundColor, 'rgba(0, 0, 0, 0)');
+
+const readMoreTextColor = getComputedStyle(readMore).color;
+assert.notStrictEqual(readMoreTextColor, 'rgba(0, 0, 0, 0)');
+
+const readMoreBorderRadius = getComputedStyle(readMore).borderRadius;
+assert.notStrictEqual(readMoreBorderRadius, '0px');
+```
+
+`.post-img` should fill the card's `width` and have a `border-bottom` value.
+
+```js
+const postImg = document.querySelector('.post-img');
+assert.exists(postImg);
+
+const imgWidth = getComputedStyle(postImg).width;
+const cardWidth = getComputedStyle(postImg.parentElement).width;
+assert.strictEqual(imgWidth, cardWidth);
+
+const imgBorderBottom = getComputedStyle(postImg).borderBottomWidth;
+assert.notStrictEqual(imgBorderBottom, '0px');
+```
+
+Ensure `.post-title` and `post-excerpt` have text color and margin.
```

Comment:
This test failed for me when I tried to use a selector like this: `.post-title, .post-excerpt`. We should allow that type of selector. Or potentially, you could test the DOM for the output to see if the elements have what you expect. Then you don't need to worry about the selectors they use. Or you could just change the user story to specify to use a `.post-title` selector, and a `.post-excerpt` selector.

Reply:
Thanks for pointing that out. I've adjusted the test to account for both grouped or individual selectors.